{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"Stdpopsim.ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"code","metadata":{"id":"PztTg8ZFR3ci"},"source":["from google.colab import drive\n","drive.mount('/content/drive')\n","%cd /content/drive/My Drive/CRISTINA DÍAZ UBIETA/Bioinformática/2nd year/Applied Bioinformatics/Google_Colab"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"f24N-eHLeIp2","executionInfo":{"status":"ok","timestamp":1604996878791,"user_tz":-60,"elapsed":31413,"user":{"displayName":"Cristina Díaz Ubieta","photoUrl":"https://lh6.googleusercontent.com/-ncg0HSFQ5Uk/AAAAAAAAAAI/AAAAAAAALFY/kfrmwRnyKAA/s64/photo.jpg","userId":"09252390706758080726"}},"outputId":"6c0d9bb9-2ed6-405a-a750-3f3421641538","colab":{"base_uri":"https://localhost:8080/"}},"source":["#@title StdPopSim\n"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","python3-testresources is already the newest version (2.0.0-2).\n","0 upgraded, 0 newly installed, 0 to remove and 12 not upgraded.\n","Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","Note, selecting 'libgsl-dev' instead of 'libgsl0-dev'\n","python-dev is already the newest version (2.7.15~rc1-1).\n","libgsl-dev is already the newest version (2.4+dfsg-6).\n","0 upgraded, 0 newly installed, 0 to remove and 12 not upgraded.\n","Requirement already satisfied: stdpopsim in /usr/local/lib/python3.6/dist-packages (0.1.2)\n","Requirement already satisfied: humanize in /usr/local/lib/python3.6/dist-packages (from stdpopsim) (0.5.1)\n","Requirement already satisfied: msprime>=0.7.1 in /usr/local/lib/python3.6/dist-packages (from stdpopsim) (0.7.4)\n","Requirement already satisfied: pyslim>=0.401 in /usr/local/lib/python3.6/dist-packages (from stdpopsim) (0.403)\n","Requirement already satisfied: attrs>=19.1.0 in /usr/local/lib/python3.6/dist-packages (from stdpopsim) (20.2.0)\n","Requirement already satisfied: appdirs in /usr/local/lib/python3.6/dist-packages (from stdpopsim) (1.4.4)\n","Requirement already satisfied: numpy>=1.7 in /usr/local/lib/python3.6/dist-packages (from msprime>=0.7.1->stdpopsim) (1.18.5)\n","Requirement already satisfied: tskit in /usr/local/lib/python3.6/dist-packages (from msprime>=0.7.1->stdpopsim) (0.3.2)\n","Requirement already satisfied: kastore in /usr/local/lib/python3.6/dist-packages (from pyslim>=0.401->stdpopsim) (0.3.1)\n","Requirement already satisfied: h5py>=2.6.0 in /usr/local/lib/python3.6/dist-packages (from tskit->msprime>=0.7.1->stdpopsim) (2.10.0)\n","Requirement already satisfied: jsonschema>=3.0.0 in /usr/local/lib/python3.6/dist-packages (from tskit->msprime>=0.7.1->stdpopsim) (3.2.0)\n","Requirement already satisfied: svgwrite>=1.1.10 in /usr/local/lib/python3.6/dist-packages (from tskit->msprime>=0.7.1->stdpopsim) (1.4)\n","Requirement already satisfied: six in /usr/local/lib/python3.6/dist-packages (from h5py>=2.6.0->tskit->msprime>=0.7.1->stdpopsim) (1.15.0)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.6/dist-packages (from jsonschema>=3.0.0->tskit->msprime>=0.7.1->stdpopsim) (50.3.2)\n","Requirement already satisfied: pyrsistent>=0.14.0 in /usr/local/lib/python3.6/dist-packages (from jsonschema>=3.0.0->tskit->msprime>=0.7.1->stdpopsim) (0.17.3)\n","Requirement already satisfied: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.6/dist-packages (from jsonschema>=3.0.0->tskit->msprime>=0.7.1->stdpopsim) (2.0.0)\n","Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/dist-packages (from importlib-metadata; python_version < \"3.8\"->jsonschema>=3.0.0->tskit->msprime>=0.7.1->stdpopsim) (3.4.0)\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"8OcnexQv-lF8"},"source":["#Importing packages and installing depencencies etc for stdpopsim\n","!sudo apt install python3-testresources\n","!sudo apt-get install python-dev libgsl0-dev\n","!python -m pip install stdpopsim "],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"KXuOBTPD-OHG","executionInfo":{"status":"ok","timestamp":1605012468645,"user_tz":-60,"elapsed":7727511,"user":{"displayName":"Cristina Díaz Ubieta","photoUrl":"https://lh6.googleusercontent.com/-ncg0HSFQ5Uk/AAAAAAAAAAI/AAAAAAAALFY/kfrmwRnyKAA/s64/photo.jpg","userId":"09252390706758080726"}},"outputId":"82e938af-53c1-41b3-e0fc-a20d36a7bc79","colab":{"base_uri":"https://localhost:8080/"}},"source":["#Running the stdpopsim. You need to manually change chromosome for each run\n","!stdpopsim DroMel -c chrX -g ComeronCrossover_dm6 -o DroMCCX.ts -d African3Epoch_1S16  4\n","#Converting the ts file into a vcf file\n","!tskit vcf DroMCCX.ts > DroMCCX.vcf"],"execution_count":null,"outputs":[{"output_type":"stream","text":["--- Logging error ---\n","Traceback (most recent call last):\n","  File \"/usr/lib/python3.6/logging/__init__.py\", line 994, in emit\n","    msg = self.format(record)\n","  File \"/usr/lib/python3.6/logging/__init__.py\", line 840, in format\n","    return fmt.format(record)\n","  File \"/usr/local/lib/python3.6/dist-packages/stdpopsim/cli.py\", line 58, in format\n","    record.args = (match.group(1),)\n","AttributeError: 'NoneType' object has no attribute 'group'\n","Call stack:\n","  File \"/usr/local/bin/stdpopsim\", line 8, in <module>\n","    sys.exit(stdpopsim_main())\n","  File \"/usr/local/lib/python3.6/dist-packages/stdpopsim/cli.py\", line 651, in stdpopsim_main\n","    run(args)\n","  File \"/usr/local/lib/python3.6/dist-packages/stdpopsim/cli.py\", line 642, in run\n","    args.runner(args)\n","  File \"/usr/local/lib/python3.6/dist-packages/stdpopsim/cli.py\", line 436, in run_simulation\n","    length_multiplier=args.length_multiplier)\n","  File \"/usr/local/lib/python3.6/dist-packages/stdpopsim/species.py\", line 141, in get_contig\n","    \"Non-autosomal simulations are not yet supported. See \"\n","  File \"/usr/lib/python3.6/warnings.py\", line 99, in _showwarnmsg\n","    msg.file, msg.line)\n","Message: '%s'\n","Arguments: ('/usr/local/lib/python3.6/dist-packages/stdpopsim/species.py:141: NonAutosomalWarning: Non-autosomal simulations are not yet supported. See https://github.com/popsim-consortium/stdpopsim/issues/383 and https://github.com/popsim-consortium/stdpopsim/issues/406\\n  \"Non-autosomal simulations are not yet supported. See \"\\n',)\n","Simulation information:\n","    Engine: msprime (0.7.4)\n","    Model id: African3Epoch_1S16\n","    Model desciption: Three epoch African population\n","    Population: number_samples (sampling_time_generations):\n","        AFR: 4 (0)\n","Contig Description:\n","    Contig length: 23009578.0\n","    Mean recombination rate: 2.9151053903465754e-08\n","    Mean mutation rate: 5.49e-09\n","    Genetic map: ComeronCrossover_dm6\n","\n","If you use this simulation in published work, please cite:\n","[stdpopsim]\n","Adrion et al., 2019: https://doi.org/10.1101/2019.12.20.885129\n","[simulation engine]\n","Kelleher et al., 2016: https://doi.org/10.1371/journal.pcbi.1004842\n","[genome assembly]\n","dos Santos et al., 2015: https://doi.org/10.1093/nar/gku1099\n","[mutation rate]\n","Schrider et al., 2013: https://doi.org/10.1534/genetics.113.151670\n","[genetic map]\n","Comeron et al, 2012: https://doi.org/10.1371/journal.pgen.1002905\n","[demographic model]\n","Sheehan and Song, 2016: https://doi.org/10.1371/journal.pcbi.1004845\n","[generation time]\n","Li et al., 2006: https://doi.org/10.1371/journal.pgen.0020166\n"],"name":"stdout"}]}]}